### PR TITLE
[SPARK-26495][SQL] Simplify the SelectedField extractor.

### PR DESCRIPTION
**Pull request by hvanhovell**
## What changes were proposed in this pull request?
The current `SelectedField` extractor is somewhat complicated and it seems to be handling cases that should be handled automatically:

- `GetArrayItem(child: GetStructFieldObject())`
- `GetArrayStructFields(child: GetArrayStructFields())`
- `GetMap(value: GetStructFieldObject())`

This PR removes those cases and simplifies the extractor by passing down the data type instead of a field.

## How was this patch tested?
Existing tests.

### CODE REVIEW

**Comment by hvanhovell** [comment]
cc @cloud-fan @rednaxelafx @mallman @dbtsai

**Review by rednaxelafx — APPROVED** [review]
LGTM. Is there any cases that might make sense to add to `SelectedFieldSuite`?

**Comment by hvanhovell** [comment]
@rednaxelafx there shouldn't be any behavioral change. It is only a simplification.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SelectedField.scala [hunk 1]
```diff
@@ -64,71 +63,51 @@ private[execution] object SelectedField {
...
-    dataType match {
-      case _: StructType =>
-        StructType(Array(field))
-      case ArrayType(elementType, containsNull) =>
-        ArrayType(wrapStructType(elementType, field), containsNull)
-      case MapType(keyType, valueType, valueContainsNull) =>
-        MapType(keyType, wrapStructType(valueType, field), valueContainsNull)
-    }
-  }
+  private def struct(field: StructField): StructType = StructType(Array(field))
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SelectedField.scala]: This seems to be very short and used only twice. Do we have future plans to use this?

Comment (hvanhovell) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/SelectedField.scala]: No, I just don't like repeating myself. I can move into StructType if you have issues with it.

**Comment by hvanhovell** [comment]
Merging this.

@dongjoon-hyun let me know if you want me to touch the struct function up.

**Pull request closed without merging**